### ISSUE TITLE
How can I position the float window on the top right with some padding?

### ISSUE THREAD

**Issue by dathinaios** [stevearc/aerial.nvim#107]
Thank you for making this, it is very useful 🙂

At the moment the floating window options seem to assume that we would like it placed at the center of the editor, window or at the cursor position. Shouldn't there be an option to position it anywhere? I would like to position it at the top right of the window with some padding. Is there maybe a way to achieve this effect with the override function?

**Comment by stevearc** [comment]
Yes, this should be doable with the override function. Try something like
```lua
require('aerial').setup({
  float = {
    relative = "editor",
    override = function(conf)
      local padding = 4
      conf.anchor = 'NE'
      conf.row = padding
      conf.col = vim.o.columns - padding
      return conf
    end,
  }
})
```

**Comment by dathinaios** [comment]
Thank you. It is working well though not with more than one splits as it is not positioned within the buffer window. But not a big issue, I have a toggle mapping to use it when I need it 🙂

**Comment by dathinaios** [comment]
I did try that but it doesn't work. It still displayes at the top right of the editor. Another approach could be to have the float at a fixed position but refreshing depending on the active buffer.

**Comment by stevearc** [comment]
Oh sorry, you'll also have to change the layout column. Instead of `vim.o.columns - padding` (which is the width of the entire editor), you'd want `vim.api.nvim_win_get_width(0) - padding`, which is the width of the current window.

**Comment by stevearc** [comment]
In general, I'd recommend checking out the documentation for `:help nvim_open_win`. It should explain what the various options do and should help you figure out what you need to tweak to get the layout you want.

**Comment by dathinaios** [comment]
It works great:

![Aerial-float](https://user-images.githubusercontent.com/481589/170485702-20d17aa8-bbff-4a55-a28e-0e7269a29d45.gif)

For reference:

```lua
  float = {
    relative = "win",
    override = function(conf)
      local padding = 1 
      conf.anchor = 'NE'
      conf.row = padding
      conf.col = vim.api.nvim_win_get_width(0) - padding
      return conf
    end,
  },
```

and the autocommand:

```lua
vim.api.nvim_exec( 
[[
    au! WinLeave *.rs AerialCloseAll
]], false)
```

I needed the autocommand because as you can see in the gif my splits resize automatically and that was messing up the placement. In any case I think this is quite elegant 🙂

Thank you for the help!

**Comment by dathinaios** [comment]
This broke with the last update. I see there is a deprecation TODO related to it in the source code but could I have some help with converting it? 😬

**Comment by stevearc** [comment]
I think this wasn't related to deprecation of anything, but instead was because of a change to when the override function gets called. I've made a tweak to that, but also I've changed the signature to pass in the source window ID. I'd recommend making the following change to your config:
```lua
  float = {
    relative = "win",
    override = function(conf, source_winid) -- <- the source_winid is new
      local padding = 1 
      conf.anchor = 'NE'
      conf.row = padding
      conf.col = vim.api.nvim_win_get_width(source_winid) - padding
      return conf
    end,
  },
```
The reason is using `0` as the window ID will use the current window, but the current window may not actually be the window you want to use for width calculations.
